projects/storefrontlib/src/lib/cms-lib/banner/banner.component.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from '@angular/core';
+
+import {
+  CmsBannerComponent,
+  CmsBannerComponentMedia,
+  CmsConfig,
+  CmsResponsiveBannerComponentMedia
+} from '@spartacus/core';
+import { CmsComponentData } from './../../cms/components/cms-component-data';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class BannerComponentService {
+  constructor(
+    public component: CmsComponentData<CmsBannerComponent>,
+    protected config: CmsConfig
+  ) {}
+
+  private convertToAbsoluteUrl = map(url => this.getBaseUrl() + url);
+
+  // TODO: move to a more generic location
+  // TODO: Make configurable
+  private formats: any[] = [
```

Comment:
Can we provide an interface here? This method should return an array of formats, not `any` imho.

Reply:
Is current version ok?

Reply:
updated